ui/hooks/gasFeeInput/useGasFeeErrors.js

```diff
@@ -267,13 +269,19 @@ export function useGasFeeErrors({
     [gasErrors, gasWarnings],
   );
 
-  const { balance: ethBalance } = useSelector(getSelectedAccount, shallowEqual);
-  const balanceError = hasBalanceError(
-    minimumCostInHexWei,
-    transaction,
-    ethBalance,
+  const account = useSelector(
+    (state) => getTargetAccount(state, transaction?.txParams?.from),
+    shallowEqual,
   );
 
+  // Balance check is only relevant for outgoing + pending transactions
+  const balanceError =
+    account !== undefined &&
+    transaction?.type !== TransactionType.incoming &&
+    transaction?.status in PENDING_STATUS_HASH
```

Comment:
I found that this balance check was being called in cases it should not.  Like for incoming transactions, and transactions that have already confirmed.  Because we won't have the `from` account in our state for incoming transactions, this guard is needed.

A better fix may be to further split these gas operations, so that callers can request only the necessary gas information for their contexts.